src/Nox.Solution/Models/Domain/Entities/Entity.cs

```diff
@@ -169,12 +170,10 @@ public void SetDefaultOtherEntityOnRelationships(Domain parentNode)
         {
             var otherEntity = grouping.Key;
             var relatedRelationships = otherEntity.Relationships.Where(r => r.Entity.Equals(Name)).ToList();
-            var count = Math.Min(grouping.Count(), relatedRelationships.Count);
-            for (var i = 0; i < count; i++)
+            foreach(var relationship in grouping)
             {
-                var relationship = grouping.ElementAt(i);
                 relationship.Related.Entity = otherEntity;
-                relationship.Related.EntityRelationship = relatedRelationships[i];
+                relationship.Related.EntityRelationship = relatedRelationships.Find(x => x.Name == relationship.RefRelationshipName) ?? relatedRelationships[0];
```

Comment:
for performancing improvement, maybe we can do this?:
```suggestion
                relationship.Related.EntityRelationship = relatedRelationships.Find(x => relationship.RefRelationshipName is not null &&  x.Name == relationship.RefRelationshipName) ?? relatedRelationships[0];
```